### DISCUSSION TITLE
Error installing latest hotfix

### DISCUSSION THREAD

**Discussion by stondino00** [Security-Onion-Solutions/securityonion#13805] [Unsupported Versions]
Would be nice to have a troubleshoot script and/or a fixer script that runs through common issues seen with Security onion and shows how to fix them.

Getting this on updating to the most recent hotfix.

Directory walk started
Directory walk done - 1891 packages
Temporary output repo path: /nsm/repo/.repodata/
Preparing sqlite DBs
Pool started (with 5 workers)
Pool finished
6 files removed
Checking for OS updates.
Looks like you have an updated OS
Applying 20241010 hotfix
tar: Removing leading `/' from member names
tar: Removing leading `/' from member names
Creating local pillar and salt directories if needed
No actions required. (2.4.110/20241010)
Hotfix applied
Updating the Security Onion version file.
Enabling highstate.
local:
    ----------
    msg:
        Info: highstate state already enabled.
    res:
        True

[INFO    ] Loading fresh modules for state activity
[INFO    ] Creating module dir '/var/cache/salt/minion/extmods/clouds'
[INFO    ] Syncing clouds for environment 'base'
[INFO    ] Loading cache from salt://_clouds, for base
[INFO    ] Caching directory '_clouds/' for environment 'base'
[INFO    ] Syncing beacons for environment 'base'
[INFO    ] Loading cache from salt://_beacons, for base
[INFO    ] Caching directory '_beacons/' for environment 'base'
[INFO    ] Copying '/var/cache/salt/minion/files/base/_beacons/zeek.py' to '/var/cache/salt/minion/extmods/beacons/zeek.py'
[INFO    ] Syncing modules for environment 'base'
[INFO    ] Loading cache from salt://_modules, for base
[INFO    ] Caching directory '_modules/' for environment 'base'
[INFO    ] Copying '/var/cache/salt/minion/files/base/_modules/healthcheck.py' to '/var/cache/salt/minion/extmods/modules/healthcheck.py'
[INFO    ] Copying '/var/cache/salt/minion/files/base/_modules/needs_restarting.py' to '/var/cache/salt/minion/extmods/modules/needs_restarting.py'
[INFO    ] Copying '/var/cache/salt/minion/files/base/_modules/so.py' to '/var/cache/salt/minion/extmods/modules/so.py'
[INFO    ] Copying '/var/cache/salt/minion/files/base/_modules/telegraf.py' to '/var/cache/salt/minion/extmods/modules/telegraf.py'
[INFO    ] Copying '/var/cache/salt/minion/files/base/_modules/zeekctl.py' to '/var/cache/salt/minion/extmods/modules/zeekctl.py'
[INFO    ] Creating module dir '/var/cache/salt/minion/extmods/states'
[INFO    ] Syncing states for environment 'base'
[INFO    ] Loading cache from salt://_states, for base
[INFO    ] Caching directory '_states/' for environment 'base'
[INFO    ] Creating module dir '/var/cache/salt/minion/extmods/sdb'
[INFO    ] Syncing sdb for environment 'base'
[INFO    ] Loading cache from salt://_sdb, for base
[INFO    ] Caching directory '_sdb/' for environment 'base'
[INFO    ] Creating module dir '/var/cache/salt/minion/extmods/grains'
[INFO    ] Syncing grains for environment 'base'
[INFO    ] Loading cache from salt://_grains, for base
[INFO    ] Caching directory '_grains/' for environment 'base'
[INFO    ] Creating module dir '/var/cache/salt/minion/extmods/renderers'
[INFO    ] Syncing renderers for environment 'base'
[INFO    ] Loading cache from salt://_renderers, for base
[INFO    ] Caching directory '_renderers/' for environment 'base'
[INFO    ] Creating module dir '/var/cache/salt/minion/extmods/returners'
[INFO    ] Syncing returners for environment 'base'
[INFO    ] Loading cache from salt://_returners, for base
[INFO    ] Caching directory '_returners/' for environment 'base'
[INFO    ] Creating module dir '/var/cache/salt/minion/extmods/output'
[INFO    ] Syncing output for environment 'base'
[INFO    ] Loading cache from salt://_output, for base
[INFO    ] Caching directory '_output/' for environment 'base'
[INFO    ] Creating module dir '/var/cache/salt/minion/extmods/utils'
[INFO    ] Syncing utils for environment 'base'
[INFO    ] Loading cache from salt://_utils, for base
[INFO    ] Caching directory '_utils/' for environment 'base'
[INFO    ] Creating module dir '/var/cache/salt/minion/extmods/log_handlers'
[INFO    ] Syncing log_handlers for environment 'base'
[INFO    ] Loading cache from salt://_log_handlers, for base
[INFO    ] Caching directory '_log_handlers/' for environment 'base'
[INFO    ] Creating module dir '/var/cache/salt/minion/extmods/executors'
[INFO    ] Syncing executors for environment 'base'
[INFO    ] Loading cache from salt://_executors, for base
[INFO    ] Caching directory '_executors/' for environment 'base'
[INFO    ] Creating module dir '/var/cache/salt/minion/extmods/proxy'
[INFO    ] Syncing proxy for environment 'base'
[INFO    ] Loading cache from salt://_proxy, for base
[INFO    ] Caching directory '_proxy/' for environment 'base'
[INFO    ] Creating module dir '/var/cache/salt/minion/extmods/engines'
[INFO    ] Syncing engines for environment 'base'
[INFO    ] Loading cache from salt://_engines, for base
[INFO    ] Caching directory '_engines/' for environment 'base'
[INFO    ] Creating module dir '/var/cache/salt/minion/extmods/thorium'
[INFO    ] Syncing thorium for environment 'base'
[INFO    ] Loading cache from salt://_thorium, for base
[INFO    ] Caching directory '_thorium/' for environment 'base'
[INFO    ] Creating module dir '/var/cache/salt/minion/extmods/serializers'
[INFO    ] Syncing serializers for environment 'base'
[INFO    ] Loading cache from salt://_serializers, for base
[INFO    ] Caching directory '_serializers/' for environment 'base'
[INFO    ] Creating module dir '/var/cache/salt/minion/extmods/matchers'
[INFO    ] Syncing matchers for environment 'base'
[INFO    ] Loading cache from salt://_matchers, for base
[INFO    ] Caching directory '_matchers/' for environment 'base'
[INFO    ] Loading fresh modules for state activity
[ERROR   ] Rendering exception occurred
Traceback (most recent call last):
  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/utils/templates.py", line 468, in render_jinja_tmpl
    output = template.render(**decoded_context)
  File "/opt/saltstack/salt/lib/python3.10/site-packages/jinja2/environment.py", line 1304, in render
    self.environment.handle_exception()
  File "/opt/saltstack/salt/lib/python3.10/site-packages/jinja2/environment.py", line 939, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "<template>", line 6, in top-level template code
  File "/opt/saltstack/salt/lib/python3.10/site-packages/jinja2/environment.py", line 1408, in make_module
    return TemplateModule(self, ctx)
  File "/opt/saltstack/salt/lib/python3.10/site-packages/jinja2/environment.py", line 1540, in __init__
    body_stream = list(template.root_render_func(context))
  File "/var/cache/salt/minion/files/base/logstash/map.jinja", line 28, in top-level template code
    {% for node_type, node_details in logstash_node_data.items() | sort %}
  File "/opt/saltstack/salt/lib/python3.10/site-packages/jinja2/sandbox.py", line 392, in call
    if not __self.is_safe_callable(__obj):
  File "/opt/saltstack/salt/lib/python3.10/site-packages/jinja2/sandbox.py", line 276, in is_safe_callable
    getattr(obj, "unsafe_callable", False) or getattr(obj, "alters_data", False)
jinja2.exceptions.UndefinedError: 'None' has no attribute 'items'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/utils/templates.py", line 211, in render_tmpl
    output = render_str(tmplstr, context, tmplpath)
  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/utils/templates.py", line 474, in render_jinja_tmpl
    raise SaltRenderError(f"Jinja variable {exc}{out}", line, tmplstr)
salt.exceptions.SaltRenderError: Jinja variable 'None' has no attribute 'items'
/var/cache/salt/minion/files/base/logstash/map.jinja(28):
---
[...]
{%       endfor %}
{%     endif %}
{%   endif %}
{% endfor %}

{% for node_type, node_details in logstash_node_data.items() | sort %}    <======================
{%   for hostname in logstash_node_data[node_type].keys() %}
{%     do LOGSTASH_NODES.append({hostname:node_details[hostname].ip}) %}
{%   endfor %}
{% endfor %}

[...]
---
[CRITICAL] Rendering SLS 'base:logstash' failed: Jinja variable 'None' has no attribute 'items'
/var/cache/salt/minion/files/base/logstash/map.jinja(28):
---
[...]
{%       endfor %}
{%     endif %}
{%   endif %}
{% endfor %}

{% for node_type, node_details in logstash_node_data.items() | sort %}    <======================
{%   for hostname in logstash_node_data[node_type].keys() %}
{%     do LOGSTASH_NODES.append({hostname:node_details[hostname].ip}) %}
{%   endfor %}
{% endfor %}

[...]
---
[INFO    ] Runner completed: 20241010215640224013
local:
    Data failed to compile:
----------
    Rendering SLS 'base:logstash' failed: Jinja variable 'None' has no attribute 'items'
/var/cache/salt/minion/files/base/logstash/map.jinja(28):
---
[...]
{%       endfor %}
{%     endif %}
{%   endif %}
{% endfor %}

{% for node_type, node_details in logstash_node_data.items() | sort %}    <======================
{%   for hostname in logstash_node_data[node_type].keys() %}
{%     do LOGSTASH_NODES.append({hostname:node_details[hostname].ip}) %}
{%   endfor %}
{% endfor %}

[...]
---

Starting crond service at 21:56:40.993664
Successfully started crond.

Starting salt-master service at 21:56:41.004957
Successfully started salt-master.

Starting salt-minion service at 21:56:41.015227
Successfully started salt-minion.

Enabling highstate.
local:
    ----------
    msg:
        Info: highstate state already enabled.
    res:
        True

Soup failed with error 1: Unhandled error

**Comment by TOoSmOotH** [comment]
That just means something was running at the same time when it was highstating. The hotfix was a small change to a script that only impacts folks with custom fleet integrations. If you run so-checkin again it should be fine.

**Reply by stondino00** [reply]
This is the state all my nodes are in since rebooting hours ago.

sudo so-status

⌛ System appears to be starting. No highstate has completed since the system
    was restarted.

**Reply by TOoSmOotH** [reply]
run the following:

`mv /opt/so/saltstack/default/salt/top.sls.backup /opt/so/saltstack/default/salt/top.sls`

Then run so-checkin
